### ISSUE TITLE
DeepFool attack returns none type while the adversarial images were generated

### ISSUE THREAD

**Issue by atomextranova** [bethgelab/foolbox#142]
My model successfully generates adversarial images on LGBFS but get None with the same code on deep fool attack.

Through debugging, I found really strange behavior
```
adv = attack(x, y) // Got None type

def call_decorator(call_fn):
    @functools.wraps(call_fn)
    def wrapper(self, input_or_adv, label=None, unpack=True, **kwargs):
            ..
           if unpack:
                  return a.image // Still has the image
```

When I use the debugger to trace back, it seems that the result was correctly stored in a.image but was unable to be returned correctly through the decorator. Not sure what the real problem is.

**Comment by atomextranova** [comment]
See the pull request for solution

**Comment by jonasrauber** [comment]
closed by #143
